### PR TITLE
Remove Deprecated Interfaces

### LINKED ISSUE #305: Remove all deprecated interfaces
Since we already removed `weight` which was only just deprecated in 0.7, let's go ahead and remove all of the older deprecated interfaces we're still carrying around.

### PR DESCRIPTION
**Pull request by ChristopherDavenport**
Removed reduce_with_identity, mul, find, and position.

Fixes #305

### CODE REVIEW

File: src/iter/mod.rs [hunk 1]
```diff
@@ -581,15 +563,6 @@ pub trait ParallelIterator: Sized {
         find_first_last::find_last(self, predicate)
     }
 
-    #[doc(hidden)]
-    #[deprecated(note = "parallel `find` does not search in order -- use `find_any`, \\
-    `find_first`, or `find_last`")]
-    fn find<P>(self, predicate: P) -> Option<Self::Item>
```
Comment (nikomatsakis) [inline src/iter/mod.rs]: Ah, wait, shoot! `find` was meant to be kept indefinitely.

Comment (nikomatsakis) [inline src/iter/mod.rs]: Would you mind reverting this part in another PR?

The idea was to help people who port over a sequential iterator and guide them to the right thing.

File: src/iter/mod.rs [hunk 2]
```diff
@@ -850,15 +823,6 @@ pub trait IndexedParallelIterator: ParallelIterator {
             .map(|(i, _)| i)
     }
 
-    #[doc(hidden)]
-    #[deprecated(note = "parallel `position` does not search in order -- use `position_any`, \\
-    `position_first`, or `position_last`")]
-    fn position<P>(self, predicate: P) -> Option<usize>
```
Comment (nikomatsakis) [inline src/iter/mod.rs]: Same with `position`

File: src/iter/mod.rs (hunk 1 above)
Comment (ChristopherDavenport) [inline src/iter/mod.rs]: I'm on it.

**Pull request merged**